x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/container/use_container_metrics_table.ts

```diff
@@ -83,33 +83,76 @@ export function useContainerMetricsTable({
 
 function seriesToContainerNodeMetricsRow(series: MetricsExplorerSeries): ContainerNodeMetricsRow {
   if (series.rows.length === 0) {
-    return {
-      name: series.id,
-      uptime: null,
-      averageCpuUsagePercent: null,
-      averageMemoryUsageMegabytes: null,
-    };
+    return rowWithoutMetrics(series.id);
   }
 
-  let uptime: number = 0;
-  let averageCpuUsagePercent: number = 0;
-  let averageMemoryUsageMegabytes: number = 0;
-  series.rows.forEach((row) => {
-    const metricValues = unpackMetrics(row);
-    uptime += metricValues.uptime ?? 0;
-    averageCpuUsagePercent += metricValues.averageCpuUsagePercent ?? 0;
-    averageMemoryUsageMegabytes += metricValues.averageMemoryUsageMegabytes ?? 0;
+  return {
+    name: series.id,
+    ...calculateMetricAverages(series.rows),
+  };
+}
+
+function rowWithoutMetrics(name: string) {
+  return {
+    name,
+    uptime: null,
+    averageCpuUsagePercent: null,
+    averageMemoryUsageMegabytes: null,
+  };
+}
+
+function calculateMetricAverages(rows: MetricsExplorerRow[]) {
+  const { uptimeValues, averageCpuUsagePercentValues, averageMemoryUsageMegabytesValues } =
+    collectMetricValues(rows);
+
+  let uptime = null;
+  if (uptimeValues.length !== 0) {
+    uptime = averageOfValues(uptimeValues);
+  }
+
+  let averageCpuUsagePercent = null;
+  if (averageCpuUsagePercentValues.length !== 0) {
+    averageCpuUsagePercent = averageOfValues(averageCpuUsagePercentValues);
+  }
+
+  let averageMemoryUsageMegabytes = null;
+  if (averageMemoryUsageMegabytesValues.length !== 0) {
+    const averageInBytes = averageOfValues(averageMemoryUsageMegabytesValues);
+    averageMemoryUsageMegabytes = Math.floor(averageInBytes / 1000000);
+  }
+
+  return {
+    uptime,
+    averageCpuUsagePercent,
+    averageMemoryUsageMegabytes,
+  };
+}
+
+function collectMetricValues(rows: MetricsExplorerRow[]) {
+  const uptimeValues: number[] = [];
+  const averageCpuUsagePercentValues: number[] = [];
+  const averageMemoryUsageMegabytesValues: number[] = [];
+
+  rows.forEach((row) => {
+    const { uptime, averageCpuUsagePercent, averageMemoryUsageMegabytes } = unpackMetrics(row);
+
+    if (uptime) {
```

Comment:
Wouldn't this truthy check discard `0` values and skew the average ?

Reply:
Right, lazy condition, I'll add the explicit null check, good catch!